### PR TITLE
[#49] [iOS] [UI] As a user, I can see emoji rating answer

### LINKED ISSUE #39: [Android] [Integrate] As a user, I can see the survey detail
## Why

Before taking a survey, users should be able to see the full preview of the details so that they can decide whether or not the survey matching their expectation.

## Acceptance Criteria

- Upon tapping the Survey Disclosure button in Home, navigate the user to Survey Details with the selected survey.
    - Pass and use the selected survey data fetched in Home.
- Use the survey title as the Title label
- Use the survey description as the Description label
- Use the survey cover image URL as the background image
    - The image is zoomed to 120% anchoring to the top right of the screen

## Design

[Survey](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=34%3A868&scaling=min-zoom)

### LINKED ISSUE #49: [iOS] [UI] As a user, I can see emoji rating answer
## Why

For Emoji Rating questions, users can rate their satisfaction by tapping on the emoji. This works like a normal rating from 1-5 stars question, but we will use emojis instead of stars

It is worth to mention that there are two Rating Question UI types.

**Smiley Rating**

Use different emojis on each rating level
It highlights only the selected one.

**Emoji Rating**

Use repeated emojis for all rating level
It highlights the selected one and all emojis on the left-hand side of the selected one.

## Acceptance Criteria
- For Emoji Rating questions,
    - Display emojis as the options
        - By default, use 👍🏻 emoji.
        - Please note that it could be other emojis as well depending on the question type.
    - Select any option should highlight every other emoji on its left
        - For instance, if a user selects the third emoji, the first, second, and third emoji should be highlighted.
    - Reduce the opacity of the non-selected range emojis

- For Smiley Rating questions,
    - Display 5 Smiley emojis as the options - 😡 😕 😐 🙂 😄
    - Select any option should highlight ONLY the selected one
        - For instance, if a user selects the third smiley, only third smiley 
should be highlighted.
    - Reduce the opacity of the non-selected emojis

## Design

[Emoji Rating Question](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=34%3A816)

### PR DESCRIPTION
**Pull request by suho**
- Close #49 

## What happened 👀

Created emoji answer view based on the display type

## Insight 📝

`N/A`

## Proof Of Work 📹

https://user-images.githubusercontent.com/19943832/211460090-6b920add-d929-4b95-a7ab-c33376193b71.mov

### CODE REVIEW

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            highlightStyle: .leftItems,
+            input: nil
+        )
+    }
+}
+
+// MARK: - EmojiAnswerView_Previews
+
+struct EmojiAnswerView_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift:96]:
```suggestion
struct EmojiAnswerView_Previews: PreviewProvider {

    static var previews: some View {
```

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            highlightStyle: .leftItems,
+            input: nil
+        )
+    }
+}
+
+// MARK: - StarEmojiAnswerView_Previews
+
+struct StarEmojiAnswerView_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift:82]:
```suggestion
struct StarEmojiAnswerView_Previews: PreviewProvider {

    static var previews: some View {
```

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        case leftItems
+        case one
+    }
+}
+
+// MARK: - ThumpsUpEmojiAnswerView_Previews
+
+struct ThumpsUpEmojiAnswerView_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift:68]:
```suggestion
struct ThumpsUpEmojiAnswerView_Previews: PreviewProvider {

    static var previews: some View {
```

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/SurveyQuestionsView.swift [hunk 4]
```diff
@@ -138,6 +138,46 @@ struct SurveyQuestionsView: View {
...
+                Spacer(minLength: 150.0)
+                EmojiAnswerView(
+                    emojis: Array(repeating: Constants.Emoji.heart, count: 5),
+                    answers: heart.answers(),
+                    inputDidChange: { _ in },
+                    highlightStyle: .leftItems,
+                    input: heart.input().first
+                )
+            }
+
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/SurveyQuestionsView.swift:163]:
```suggestion
```

**Review by nkhanh44 — APPROVED** [review]
So good so good 🚀

File: ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            Spacer()
+        }
+    }
+
+    func isHighlighted(for index: Int) -> Bool {
+        switch highlightStyle {
+        case .leftItems:
+            return index <= currentIndex
+        case .one:
+            return index == currentIndex
```
Comment (vnntsu) [inline ios/Surveys/Sources/Presentation/Modules/SurveyQuestions/Views/EmojiAnswerView.swift:49]:
```suggestion
        case .leftItems: return index <= currentIndex
        case .one: return index == currentIndex
```

**Review by vnntsu — APPROVED** [review]
Just minor comments. All good 🚀

**Pull request merged**